docs/_build/html/_sources/methods.rst.txt

```diff
@@ -0,0 +1,444 @@
+Reversal potential      ``E``
+Inactivation variable   ``h``
+Activation variable     ``m``
+======================= ================
+
+For example ::
+
+  x.HH.add('prinz/CaS', 'gbar', 500, 'E', -80);
+
+The maximal conductance defaults to 0, the reversal potential defaults based on
+the conductance. Activation variables (if any) default to zero. Inactivation variables
+default to one. Synapses can be added this way, but it is better to use the ``connect`` method.
+
+.. _cleanup:
+
+cleanup
+^^^^^^^
+
+When the ``C++`` code compiles, it produces a binary ``.mexa64`` file and a ``.cpp`` file in your ``xolotl`` directory. These files are hashed so that repeated simulation does not require recompilation. If these files are too numerous or broken, you can erase them all by calling either ::
+
+  xolotl.cleanup
+  x.cleanup
+
+
+.. _compile:
+
+compile
+^^^^^^^
+
+Hashes and compiles the files needed to run a simulation. These are stored in
+your ``xolotl`` directory. ``xolotl`` automatically compiles when it needs to.
+You can turn this functionally off by setting ::
+
+  x.skip_hash = true;
+
+In addition, creating a ``xolotl`` object through a function call does not
+automatically hash and compile. In this case, you should use ``x.md5hash``.
+=======
+In addition, creating a ``xolotl`` object through a function call does not automatically hash and compile. In this case, you should use ``x.md5hash``.
+>>>>>>> 2c44bd81aed5d11aac7fb1677a4b462d47822744
+
+.. hint::
+
+  Always transpile before you compile! ::
+
+    x.transpile;
+    x.compile;
+
+.. _connect:
+
+connect
+^^^^^^^
+
+Connects two compartments with a synapse. The basic syntax is ::
+
+  x.connect('PreSynaptic', 'PostSynaptic', 'Type', PropertyName', PropertyValue, ...)
+
+The first two arguments are the presynaptic and postsynaptic compartment names.
+
+In the case of two arguments, or three arguments where the third is a ``double``,
+an axial or electrical synapse is created between the two compartments. An axial
+synapse is created if either of the compartments is part of a spacially-discretized
+multi-compartment structure (e.g. has a defined ``tree_idx``). Otherwise, the created
+synapse is electrical. Axial and electrical synapses differ in how they are integrated
+(see Dayan & Abbott 2001, Ch. 5-6). ::
+
+  % create a synapse between AB and LP with gbar of NaN
+  x.connect('AB', 'LP')
+  % create a synapse between AB and LP with gbar of 10
+  x.connect('AB', 'LP', 10)
+
+The most common way to produce a synapse is to pass the synapse type and then any
+properties. For example, to add a glutamatergic synapse between ``AB`` and ``LP``
+with a maximal conductance of 100: ::
+
+  x.connect('AB', 'LP', 'Glut', 'gbar', 100)
+
+
+Synapses can also be connected by passing a ``cpplab`` object to the ``connect``
+function ::
+
+  % create a synapse using the cpplab object 'syn_cpplab'
+  x.connect('AB', 'LP', syn_cpplab)
+
+The following properties can be specified
```

Comment:
i wouldn't emphasize this -- it's possible for the user to create their own synapse type, which will have whatever props they add